consensus/src/chained_bft/network.rs

```diff
@@ -445,11 +444,11 @@ impl<T: Payload> NetworkTask<T> {
                             .log();
                         e
                     })?;
-                Ok(self.vote_tx.push(peer_id, vote_msg)?)
+                self.vote_tx.push(peer_id, vote_msg)
             }
-            Ordering::Greater => Ok(self
-                .future_epoch_tx
-                .push(peer_id, (vote_msg.epoch(), peer_id))?),
+            Ordering::Less | Ordering::Greater => self
```

Comment:
The code of epoch comparison seems to be repetitive for all message types: do you think we could share this logic across message types?

Reply:
sure, I'll try to do it in another PR